src/ScriptCoverageManager.cc

```diff
@@ -127,31 +124,47 @@ bool ScriptCoverageManager::WriteStats()
 		return false;
 		}
 
-	for ( auto s : stmts )
+	for ( auto& s : stmts )
 		{
-		ODesc location_info;
-		s->GetLocationInfo()->Describe(&location_info);
 		ODesc desc_info;
 		s->Describe(&desc_info);
-		string desc(desc_info.Description());
-		canonicalize_desc cd{delim};
-		for_each(desc.begin(), desc.end(), cd);
-		pair<string, string> location_desc(location_info.Description(), desc);
-		if ( usage_map.find(location_desc) != usage_map.end() )
-			usage_map[location_desc] += s->GetAccessCount();
-		else
-			usage_map[location_desc] = s->GetAccessCount();
+		TrackUsage(s, desc_info.Description(), s->GetAccessCount());
 		}
 
-	map<pair<string, string>, uint64_t>::const_iterator it;
-	for ( auto& um : usage_map )
+	for ( auto& [func, body] : func_instances )
 		{
-		fprintf(f, "%" PRIu64 "%c%s%c%s\n", um.second, delim, um.first.first.c_str(), delim,
-		        um.first.second.c_str());
+		auto ft = func->GetType<FuncType>();
+		auto desc = ft->FlavorString() + " " + func->Name() + " TOTAL";
```

Comment:
I'm not sure about "TOTAL" here. That makes me think that it's a "total usage" count of some sort and then wonder where the actual value is. I don't have any better ideas though. Maybe "FULL"?

Reply:
Yeah I wondered about that too. I wanted _some_ simple way of pulling out the function lines (it also works to grep on _`<tab>`_`(event|hook|function)` but that's certainly ugly).

How about `BODY`, given it's the coverage count for exactly one body? (An event handler, for example, might have several of these - each with different location information of course.)

Reply:
Yeah, that sounds better 👍🏼